src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -1649,9 +1656,22 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End If
         End Sub
 
+        Friend Function ShouldAddEvent(symbol As Symbol) As Boolean
+            If EventQueue Is Nothing Then
+                Return False
+            End If
+
+            For Each location As Location In symbol.Locations
+                If location.SourceTree?.IsEmbeddedOrMyTemplateTree <> True Then
```

Comment:
I believe that, if location.SourceTree IsNot Nothing, then IsEmbeddedOrMyTemplateTree is guaranteed to be False for that tree. Have you seen examples to the contrary? Otherwise, consider replacing this condition with "location.SourceTree IsNot Nothing" followed by an assert "Debug.Assert(AllSyntaxTrees.Contains(location.SourceTree))" within the 'If'.